search-app/test/cmr/search/test/unit/services/query_execution.clj

```diff
@@ -3,116 +3,143 @@
+            {:concept-id ["G1-PROV1" "G2-PROV1"]
+             :page-size 2
+             :result-format :opendata}
+            :specific-elastic-items
+
+            {:concept-id ["G1-PROV1" "G2-PROV1"]
+             :page-size 1
+             :result-format :opendata}
+            :elasticsearch
+
+            ;; A different page number forces elastic strategy
+            {:concept-id ["G1-PROV1" "G2-PROV1"] :result-format :opendata :page-num 2} :elasticsearch
+
+            ;; XML References use elastic strategy
+            {:concept-id "G1-PROV1" :result-format :xml} :elasticsearch
+
+            ;; Sorting uses the elastic strategy
+            {:concept-id ["G1-PROV1" "G2-PROV1"]
+             :result-format :echo10
+             :sort-key ["concept-id"]}
+            :elasticsearch
+
+            {:concept-id ["G1-PROV1" "G2-PROV1"]
+             :page-size 1
+             :result-format :echo10}
+            :elasticsearch
+
+            ;; A different page number forces elastic strategy
+            {:concept-id ["G1-PROV1" "G2-PROV1"] :result-format :echo10 :page-num 2} :elasticsearch))))
+
+    (testing "Collection Specific"
+      (are [params expected-strategy]
+           (is (= expected-strategy (params->query-execution-strategy :collection params)))
+
+        ;; Facets are supported
+        {:concept-id "G1-PROV1" :result-format :opendata :include-facets "true"} :specific-elastic-items
+
+        ;; Facets require elastic strategy
+        {:concept-id "G1-PROV1" :result-format :echo10 :include-facets "true"} :elasticsearch
+
+        ;; All revisions uses elastic strategy
+        {:concept-id "C1-PROV1" :result-format :opendata :all-revisions "true"} :elasticsearch
+        {:concept-id "C1-PROV1" :result-format :echo10 :all-revisions "true"} :elasticsearch
 
       ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-      ;; Metadata  queries
-      {:concept-id "C1-PROV1" :result-format :echo10} :elasticsearch
-      {:concept-id "C1-PROV1" :result-format :dif} :elasticsearch
-      {:concept-id "C1-PROV1" :result-format :dif10} :elasticsearch
-      {:concept-id "C1-PROV1" :result-format :iso19115} :elasticsearch
-      {:concept-id "C1-PROV1" :result-format :iso-smap} :elasticsearch))
+        ;; Metadata  queries
+        {:concept-id "C1-PROV1" :result-format :echo10} :elasticsearch
+        {:concept-id "C1-PROV1" :result-format :dif} :elasticsearch
+        {:concept-id "C1-PROV1" :result-format :dif10} :elasticsearch
+        {:concept-id "C1-PROV1" :result-format :iso19115} :elasticsearch
+        {:concept-id "C1-PROV1" :result-format :iso-smap} :elasticsearch))
 
-  (testing "Granule Specific"
-    (are [params expected-strategy]
-         (is (= expected-strategy (params->query-execution-strategy :granule params)))
+    (testing "Granule Specific"
+      (are [params expected-strategy]
```

Comment:
I think your editor changed the spacing so not asking you to change this.   Otherwise I would ask you to use are3.